File: Python/8py 1/cube_checker.py
"""
|--------------------------------------------------------------------------
| Find out whether the shape is a cube
|--------------------------------------------------------------------------
|
"""

def cube_checker(volume, side):
   return volume > 0 and side > 0 and side ** 3 == volume


def cube_checker(volume, side):
   return (side > 0 < volume) and volume == (side**3)


def cube_checker(volume, side):
   return side > 0 and side ** 3 == volume


print(cube_checker(-12,2), False)
print(cube_checker(8, 3),  False)
print(cube_checker(8, 2),  True)
print(cube_checker(-8,-2), False)
print(cube_checker(0, 0),  False)

File: PHP/8php 1/cubeChecker.php
<?php
/*
|--------------------------------------------------------------------------
| Find out whether the shape is a cube
|--------------------------------------------------------------------------
|
*/

function cubeChecker($volume, $side){
   return $side > 0 && $side ** 3 == $volume;
}


function cubeChecker($volume, $side){
   return $volume == pow($side, 3) >= 0 ;
}


function cubeChecker($volume, $side) {
   return $volume > 0 && $volume == $side ** 3;
}


print_r(cubeChecker(56.3, 1) . ' ' . false . '
');
print_r(cubeChecker(-1, 2) . ' ' . false . '
');
print_r(cubeChecker(8, 3) . ' ' . false . '
');
print_r(cubeChecker(8, 2) . ' ' . true . '
');
print_r(cubeChecker(-8, -2) . ' ' . false . '
');
print_r(cubeChecker(0, 0) . ' ' . false . '
');